lib/kennel/parts_writer.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+require "find"
+
+module Kennel
+  class PartsWriter
+    def initialize(filter:)
+      @filter = filter
+    end
+
+    def write(parts)
+      Progress.progress "Storing" do
+        if filter.tracking_id_filter
+          write_changed(parts)
+        else
+          old = old_paths
+          used = write_changed(parts)
+          (old - used).each { |p| FileUtils.rm_rf(p) }
+        end
+      end
+    end
+
+    private
+
+    attr_reader :filter
+
+    def write_changed(parts)
+      used = ["generated"]
+
+      Utils.parallel(parts, max: 2) do |part|
+        path = "generated/#{part.tracking_id.tr("/", ":").sub(":", "/")}.json"
+        used.concat [File.dirname(path), path] # only 1 level of sub folders, so this is safe
+        payload = part.as_json.merge(api_resource: part.class.api_resource)
+        write_file_if_necessary(path, JSON.pretty_generate(payload) << "\n")
+      end
+
+      used
+    end
+
+    def apply_cleanup_to
```

Comment:
`apply_to` sounds like it's doing an action
```suggestion
    def directory_for_cleanup
```

Reply:
6cb0ed1a1eb2d0015596b5004f8bc4febcf013a2